perceval/backends/core/googlehits.py

```diff
@@ -0,0 +1,251 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2015-2018 Bitergia
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA.
+#
+# Authors:
+#     Valerio Cosentino <valcos@bitergia.com>
+#
+
+import bs4
+import hashlib
+import logging
+import re
+
+from grimoirelab_toolkit.datetime import datetime_utcnow
+from grimoirelab_toolkit.uris import urijoin
+
+from ...backend import (Backend,
+                        BackendCommand,
+                        BackendCommandArgumentParser)
+from ...client import HttpClient
+
+# Default sleep time and retries to deal with connection/server problems
+DEFAULT_SLEEP_TIME = 1
+MAX_RETRIES = 5
+
+CATEGORY_HITS = "google_hits"
+GOOGLE_SEARCH_URL = 'https://www.google.com/search'
+
+
+logger = logging.getLogger(__name__)
+
+
+class GoogleHits(Backend):
+    """GoogleHits backend for Perceval.
+
+    This class retrieves the list of hits for a given set of keywords
+    obtained from the Google API. To initialize this class a list of
+    keywords is needed.
+
+    :param keywords: a list of keywords
+    :param tag: label used to mark the data
+    :param archive: archive to store/retrieve items
+    :param max_retries: number of max retries to a data source
+        before raising a RetryError exception
+    :param sleep_time: time to sleep in case
+        of connection problems
+    """
+    version = '0.1.0'
+
+    CATEGORIES = [CATEGORY_HITS]
+
+    def __init__(self, keywords, tag=None, archive=None,
+                 max_retries=MAX_RETRIES, sleep_time=DEFAULT_SLEEP_TIME):
+        # sort keywords to produce always the same hash for the same keywords
+        keywords.sort()
```

Comment:
This sorts the original list. Code must not change inputs when the user doesn't expect it. It would be better to copy the list into `self.keywords` and sort this one.